### DISCUSSION TITLE
How to use GraphRAG with existing Cosmos DB NoSQL data?

### DISCUSSION THREAD

**Discussion by jnicholls82** [Azure-Samples/graphrag-accelerator#200] [Q&A] [answered]
Hi,

We have a Cosmos DB NoSQL database containing millions of rows of data, which we have chunked in a way that makes sense for our application. 

I would love to be able to use this as the source of data which GraphRAG would index.

Is this possible? If not supported out-of-the-box, which components would we need to edit?

Thanks,
James

**Comment by kallsu** [comment, marked as answer]
Hi,

I hope that this response will help you.

> We have a Cosmos DB NoSQL database containing millions of rows of data, which we have chunked in a way that makes sense for our application.

If your chunk can work with the index created, then the concept out-of-the-box works, otherwise, this project is an "Accelerator" that provides you the component in the base form.

According to my opinion, to avoid the re-indexing of the entire DB, let your team focus on the part of the indexing and update that part as you like it.

Starting point for your work:

- The index "model" is [here](https://github.com/Azure-Samples/graphrag-accelerator/blob/main/backend/src/api/index.py)  
- The query part is [here](https://github.com/Azure-Samples/graphrag-accelerator/blob/main/backend/src/api/query.py) and [here](https://github.com/Azure-Samples/graphrag-accelerator/blob/main/backend/src/api/query_streaming.py)

Thanks
